### PR TITLE
Test with python 3.12 and 3.13 and use 3.13 for the other github actions

### PR DESCRIPTION
**Pull request by jwillemsen**
* .devcontainer/devcontainer.json:
    * .github/workflows/automatic_generation.yml:
    * .github/workflows/generate_package.yml:
    * .github/workflows/make_release.yml:
    * .github/workflows/publish.yml:
    * .github/workflows/pytest.yml:
    * setup.py:

### CODE REVIEW

**Review by brianegge — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
     version=__version__,
     packages=find_packages(where="src"),
     package_dir={"": "src"},
-    python_requires=">=3.12",
+    python_requires=">=3.13",
```
Comment (MatthewFlamm) [inline setup.py:22]: From my reading, homeassistant still supports 3.12 and still tests against it 

See

https://github.com/home-assistant/core/blob/24f7bae5f260c7b4417e0b21d4f742ca916b2349/pyproject.toml#L25

https://github.com/home-assistant/core/blob/24f7bae5f260c7b4417e0b21d4f742ca916b2349/.github/workflows/ci.yaml#L45

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.12"]
+        python-version: ["3.13"]
```
Comment (MatthewFlamm) [inline .github/workflows/pytest.yml:22]: We should test against 3.12 and 3.13

File: .github/workflows/automatic_generation.yml [hunk 3]
```diff
@@ -59,7 +59,7 @@ jobs:
     if: needs.generate_package.outputs.need_to_release == 'true'
     strategy:
       matrix:
-        python-version: ['3.12']
+        python-version: ['3.13']
```
Comment (MatthewFlamm) [inline .github/workflows/automatic_generation.yml:62]: We should test against 3.12 and 3.13

**Review by MatthewFlamm — CHANGES_REQUESTED** [review]

**Review by MatthewFlamm — APPROVED** [review]
Thanks!

**Pull request merged**